### PR TITLE
Removed some compatibility bits for Python 3.2.

### PR DESCRIPTION
**Pull request by timgraham**

### CODE REVIEW

**Comment by charettes** [comment]
This got me wondering if we're planing to set up something in order to make sure we don't introduce `u` string literals now that we drop Python3.2 support. They can be easy to miss while reviewing a patch and a failure on Python3.2 was a great way of spotting them.

**Comment by timgraham** [comment]
That seems like something that could be implemented as part of flake8. I haven't looked to see if it might exist.

**Comment by berkerpeksag** [comment]
Here's a couple of candidate for the PR:

* https://github.com/django/django/blob/master/django/core/mail/message.py#L107
* https://github.com/django/django/blob/master/docs/faq/install.txt#L19
* https://github.com/django/django/blob/master/docs/intro/install.txt#L13
* https://github.com/django/django/blob/master/docs/topics/install.txt#L11 (there's a typo in that line: 3.3 should be 3.3+)

File: django/utils/html_parser.py [hunk 1]
```diff
@@ -4,10 +4,7 @@
 
 current_version = sys.version_info
 
-use_workaround = (
-    (current_version < (2, 7, 3)) or
-    (current_version >= (3, 0) and current_version < (3, 2, 3))
-)
+use_workaround = current_version < (2, 7, 3)
```
Comment (claudep) [inline django/utils/html_parser.py]: We might question the support of  Python < 2.7.3. I've read that Mac OS X Lion ships 2.7.1. Would this be a blocker?

Comment (timgraham) [inline django/utils/html_parser.py]: I don't feel strongly either way in this case. If there were a maintenance overhead here I'd definitely be in favor of enforcing our "we only support the latest minor release" more aggressively.

Comment (claudep) [inline django/utils/html_parser.py]: Let's keep that, you're right that this doesn't cost us much.

**Pull request merged**